ui/pages/bridge/utils/priceImpact.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Formats price impact percentage for display.
+ *
+ * @param priceImpact - The price impact value from the API (in decimal form, e.g., 0.87 = 87%).
+ * @returns Formatted price impact string.
+ */
+export function formatPriceImpact(
+  priceImpact: string | number | undefined,
+): string {
+  if (priceImpact === undefined || priceImpact === null) {
+    return '0%';
+  }
+
+  const impact = Number(priceImpact);
+
+  // If the impact is 0, show 0%
+  if (impact === 0) {
+    return '0%';
+  }
+
```

Comment:
We need to return early if the number is NaN. This also dedupes the 0% conditions

```suggestion
  const impact = Number(priceImpact);
  if (isNaN(impact) || !impact) {
    return '0%';
  }

```